urllib3/connectionpool.py

```diff
@@ -687,12 +687,17 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                 retries = retries.increment(method, url, response=response, _pool=self)
             except MaxRetryError:
                 if retries.raise_on_redirect:
-                    # Release the connection for this response, since we're not
-                    # returning it to be released manually.
+                    # Drain and release the connection for this response, since
+                    # we're not returning it to be released manually.
+                    response.read()
                     response.release_conn()
```

Comment:
Nope. =) This still represents an API extension, and additionally represents that kind of API wart which is a method that totally varies in its behaviour based on a boolean flag. Always better to have two methods than a method like that.